src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts

```diff
@@ -205,40 +226,38 @@ export class DiscussionThread {
         this.isLoading[this.dealDiscussion.createdBy.address] = false;
       });
 
-    if (this.threadComments && this.dealDiscussion) {
-      this.subscribeToDiscussion(discussionId);
-
-      if (!this.threadComments || !Object.keys(this.threadComments).length) return;
-
-      // Dictionary is used for replies, to easily find the comment by its id
-      this.threadDictionary = this.arrayToDictionary(this.threadComments);
-
-      /* Comments author profiles */
-      this.threadComments.forEach((comment: IComment) => {
-        if (!this.threadProfiles[comment.author]) {
-          this.isLoading[comment.author] = true;
-          if (comment.authorENS /* author has ENS name */) {
-            this.threadProfiles[comment.author] = {
-              name: comment.authorENS,
-              address: comment.author,
-              image: "",
-            };
-          } else {
-            /* required for replies */
-            this.discussionsService.loadProfile(comment.author).then(profile => {
-              this.threadProfiles[comment.author] = profile;
-              this.isLoading[comment.author] = false;
-            });
-          }
+    this.subscribeToDiscussion(discussionId);
+
+    if (!this.threadComments || !Object.keys(this.threadComments).length) return;
+
+    // Dictionary is used for replies, to easily find the comment by its id
+    this.threadDictionary = this.arrayToDictionary(this.threadComments);
+
+    /* Comments author profiles */
+    this.threadComments.forEach((comment: IComment) => {
+      if (!this.threadProfiles[comment.author]) {
+        this.isLoading[comment.author] = true;
+        if (comment.authorENS /* author has ENS name */) {
+          this.threadProfiles[comment.author] = {
+            name: comment.authorENS,
+            address: comment.author,
+            image: "",
+          };
+        } else {
+          /* required for replies */
+          this.discussionsService.loadProfile(comment.author).then(profile => {
+            this.threadProfiles[comment.author] = profile;
+            this.isLoading[comment.author] = false;
+          });
         }
-      });
+      }
+    });
 
-      // Update the discussion status
-      this.updateDiscussionListStatus(
-        new Date(parseFloat(this.threadComments[this.threadComments.length - 1].createdOn)),
-        this.threadComments.length,
-      );
-    }
+    // Update the discussion status
+    this.updateDiscussionListStatus(
+      new Date(parseFloat(this.threadComments[this.threadComments.length - 1].createdOn)),
+      this.threadComments.length,
+    );
```

Comment:
This whole block, I just removed the surrounding if statement `    if (this.threadComments && this.dealDiscussion) {` (see red diff line 208)

Reason: Moved the if check earlier